common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -734,15 +733,18 @@ def get_youtube_link(video_id):
             # Get video encodings for val profiles.
             val_video_encodings = edxval_api.get_urls_for_profiles(self.edx_video_id, val_profiles)
 
-            # If multiple encodings are there in val, the priority will be: youtube > hls > mp4 and webm.
+            # priority will be given to VAL's youtube source.
             if val_video_encodings.get('youtube'):
                 source_url = self.create_youtube_url(val_video_encodings['youtube'])
-            elif val_video_encodings.get('hls'):
-                source_url = val_video_encodings['hls']
-            elif val_video_encodings.get('desktop_mp4'):
-                source_url = val_video_encodings['desktop_mp4']
-            elif val_video_encodings.get('desktop_webm'):
-                source_url = val_video_encodings['desktop_webm']
+
+            # If no youtube source is provided externally or in VAl, update source_url in order: hls > mp4 and webm
+            if not source_url:
```

Comment:
I've not tested it but this does't looks good to me. There is no differentiation between external url vs the default url. 
Note: We've to override the default url according to the priority. Also, if there is external youtube url that would get the highest priority when there is not YT encoding in VAL. 
Also, when there is external YT url along with VAL YT encoding then VAL's YT will get the priority.

Reply:
That is exactly how it is implemented here. At line [723](https://github.com/edx/edx-platform/pull/15870/files#diff-2386122d0b4952fd587ad8fa9556ef8dR723) `source_url` is assigned to external YT encoding. Next at line [737](https://github.com/edx/edx-platform/pull/15870/files#diff-2386122d0b4952fd587ad8fa9556ef8dR737) `source_url` is assigned to VAL YT encoding if it is available.
At this point if `source_url` has some value then it essentially means there is either external YT or VAL YT in it. If no `source_url` we can proceed with other encodings from VAL according to priority. 
Does it make sense?